R/get_biomart_pfam_merge.R

```diff
@@ -0,0 +1,120 @@
+#' Given pfam id(s) and an ensembl mart
+#' creates a grange for the pfam domains
+#' 
+#' @param ensembl a S4 class from biomaRt::useMart() call
+#' @param pfamDesc_path path to pfam description file
+#' @param ucscGenePfam_path path to ucsc gene pfam genomic location
+#' @param pfam_id pfam id(s) to filter
+#' @param return_pfam_gr logical value to return a pfam location granges instead of  dataframe
+#' 
+#' @export
+#'
+#' @return granges for genomic locations of given pfam id(s)
+#'
+#' @examples
+#' \dontrun{
+#' # hsapiens_gene_ensembl == Human genes (GRCh38.p13) 10th Feb 2021
+#' ensembl <- biomaRt::useMart(biomart = "ENSEMBL_MART_ENSEMBL", dataset = "hsapiens_gene_ensembl", host = "ensembl.org")
+#'
+#' download.file("http://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/pfamDesc.txt.gz", "pfamDesc.txt.gz")
+#' download.file("http://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/ucscGenePfam.txt.gz", "ucscGenePfam.txt.gz")
```

Comment:
I think these should best be handled with a mechanism such as BiocFileCache to avoid re-downloading or similar. Can you please look into that?

In the end: it would download if not there yet, or point to the local stored version if it is there.

Alternative: is this something that can also be found via AnnotationHub? Because this can be related to other comments below

Reply:
(I mean, these are quite small files, so it does not make too much of a difference, but I guess it can play a role when it gets reviewed)